components/suggestion/command_provider/app_command_parser/app_command_parser.ts

```diff
@@ -0,0 +1,1256 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+/* eslint-disable max-lines */
```

Comment:
(3/5) There's a ton going on in this file, so could we maybe split it up into a few files? I know the warning is annoying with existing files that are thousands of lines long for legacy reasons, but I'd prefer to not add new files that have that problem

Reply:
@levb There's currently two classes in the file at the moment. Low-hanging fruit would be to move `ParsedCommand` to its own file. Anything else you can think of to extract from the file?

Reply:
All this logic sort of belongs together, I am slightly reluctant to break it up. Will wait for other feedback from @hmhealey first.